homeassistant/components/litterrobot/entity.py

```diff
@@ -36,18 +36,17 @@ class LitterRobotEntity(
 
     _attr_has_entity_name = True
 
-    def __init__(self, robot: _RobotT, entity_type: str, hub: LitterRobotHub) -> None:
+    def __init__(
+        self, robot: _RobotT, hub: LitterRobotHub, description: EntityDescription
+    ) -> None:
         """Pass coordinator to CoordinatorEntity."""
         super().__init__(hub.coordinator)
         self.robot = robot
-        self.entity_type = entity_type
         self.hub = hub
-        self._attr_name = entity_type.capitalize()
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID."""
-        return f"{self.robot.serial}-{self.entity_type}"
+        self.entity_description = description
+        self._attr_unique_id = f"{self.robot.serial}-{description.name}"
```

Comment:
We shouldn't use the description name for the unique id. Let's fix this now.

Reply:
Thanks, I was just trying to keep the PR small and make incremental changes to get there since this matched previous logic. I'll add it now though since I had it saved in a stash!